metroscore/service_areas.py

```diff
@@ -104,6 +105,76 @@ def time_dependent_dijkstra(
     return L
 
 
+def time_dependent_a_star(
+    G: MultiGraph, timetable: pd.Series, start_time: float, origin_id: Any
+) -> Dict[int, float]:
+    """
+    Implements a modified A* algorithm that finds the time-dependent shortest path from
+    origin node to all other nodes in the graph.
+
+    Args:
+        G (Graph): NetworkX Graph object.
+        timetable (pd.Series): Departure times. Index is a tuple of (from, to)
+            node IDs. Values are in seconds (since midnight).
+        start_time (float): Current time.
+        origin_id (Any): Origin node ID.
+
+    Returns:
+        Dict[int, float]: Dictionary of node IDs and arrival times, in seconds.
+    """
+
+    def _zero_heuristic(G: MultiGraph, target_id: Any) -> Dict[Any, float]:
+        heuristic = {}
+        for node in G.nodes:
+            heuristic[node] = 0.0
+        return heuristic
```

Comment:
nit: let's put this in utils in the future, and pass in the heuristic function as a parameter to `time_dependent_a_star`. That way we can more easily support future heuristic functions.